### ISSUE TITLE
[Feature request]: execute `loadTranslation$` when use `changeLocale` on a non declared locale

### ISSUE THREAD

**Issue by ahnpnl** [robisim74/qwik-speak#22]
### Problem

- Given speak config below
```
export const speakConfig: SpeakConfig = {
    defaultLocale: { lang: 'en-US' },
    supportedLocales: [{ lang: 'en-US' }],
    assets: ['common'],
};
```

- Given `loadTranslation$` like below
```
export const loadTranslation$: LoadTranslationFn = $(async (lang: string, asset: string, origin?: string) => {
    let url = '';
    // Absolute urls on server
    if (isServer && origin) {
        url = origin;
    }
    url += `/assets/i18n/${lang}/${asset}.json`;
    let data: object | null = null;

    try {
        const response = await fetch(url);
        data = await response.json();
    } catch (error) {
        // Implement error handling here
        console.log('loadTranslation$ error: ', error);
    }

    return data;
});
```

- Given the codes which use `changeLocale` like below
```
export const Header = component$(() => {
    const ctx = useSpeakContext();

    return (
        <header>
            <div className={'header-banner'}></div>
            <button onClick$={() => changeLocale({ lang: 'nl-NL' }, ctx)}>Change locale</button>
        </header>
    );
});
```

What I noticed was, when I clicked the button to execute `changeLocale`, the `loadTranslation$` didn't execute which took me a while to notice that my speak config didn't include locale `nl-NL`. 

### Suggestion

Can we make `loadTranslation$` execute in this case so that the `catch` can log something?

**Comment by robisim74** [comment]
Hi @ahnpnl ,

as I said [here](https://github.com/robisim74/qwik-speak/commit/f7d5920d787b2dd42d6b77f2fbae705fa5c2c374#r99698876), and as explained in [doc](https://robisim74.gitbook.io/qwik-speak/library/quick-start#change-locale), now `changeLocale` fall back to default locale if the new locale is not in `supportedLocales` (as during the first loading). An exception in `loadTranslation$` is thrown only if the file is not found, and not the locale.

However I think I will add a logger in future releases to report these cases in _dev_ mode (I still have to check if the new Qwik `isDev` function works in external libraries.

**Comment by robisim74** [comment]
I tried the new Qwik function `isDev` and it works properly: the code is removed in prod mode. So in v0.6.2 I added a logger in dev mode, to notify:
- the resolved locale when app starts
- the new locale when language changes
- fallback to default locale if the language is not supported

Please let me know if this is helpful, and if any other notifications might be needed

**Comment by ahnpnl** [comment]
Thanks a lot, I think for now these loggings are super helpful 👍 

I will close this issue as already implemented in **0.6.2**
